File: product-service/src/main/java/com/groupproject/productservice/client/ImageClient.java
package com.groupproject.productservice.client;


import com.groupproject.productservice.model.request.ImageLinksRequest;
import com.groupproject.productservice.model.response.ImageLinksResponse;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

@FeignClient(serviceId = "image-service")
public interface ImageClient {
    @RequestMapping(method = RequestMethod.POST, value = "/image/links")
    @ResponseBody
    ImageLinksResponse getLinks(ImageLinksRequest request);

}

File: api-gateway/src/main/kotlin/com/groupproject/apigateway/client/ImageClient.kt
package com.groupproject.apigateway.client

import com.groupproject.apigateway.domain.image.ImageLinksRequest
import com.groupproject.apigateway.domain.image.ImageLinksResponse
import org.springframework.cloud.netflix.feign.FeignClient
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.ResponseBody

@FeignClient(serviceId = "image-service")
interface ImageClient {

    @RequestMapping(method = arrayOf(RequestMethod.POST), value = "/image/links")
    @ResponseBody
    fun getLinks(request: ImageLinksRequest): ImageLinksResponse
}